precompile/contracts/juror/limit_orders.go

```diff
@@ -61,41 +62,59 @@ func ValidatePlaceLimitOrder(bibliophile b.BibliophileClient, inputStruct *Valid
 		orderSide = Side(Short)
 	}
 	if order.ReduceOnly {
+		// a reduce only order should reduce position
 		if !reducesPosition(posSize, order.BaseAssetQuantity) {
-			response.Errs = ErrReduceOnlyBaseAssetQuantityInvalid.Error()
+			response.Err = ErrReduceOnlyBaseAssetQuantityInvalid.Error()
 			return
 		}
 		longOrdersAmount := bibliophile.GetLongOpenOrdersAmount(trader, order.AmmIndex)
 		shortOrdersAmount := bibliophile.GetShortOpenOrdersAmount(trader, order.AmmIndex)
-		if (orderSide == Side(Long) && longOrdersAmount.Sign() != 0) || (orderSide == Side(Short) && shortOrdersAmount.Sign() != 0) {
-			response.Errs = ErrOpenOrders.Error()
+
+		// if the trader is placing a reduceOnly long that means they have a short position
+		// we allow only 1 kind of order in the opposite direction of the position
+		// otherwise we run the risk of having stale reduceOnly orders (orders that are not actually reducing the position)
+		if (orderSide == Side(Long) && longOrdersAmount.Sign() != 0) ||
```

Comment:
this check is not required in ValidatePlaceIOCorder?

Reply:
nope, will explain on call why